src/attributes/codegen.md

```diff
@@ -143,6 +143,54 @@ feature detection on the x86 platforms.
 > may be enabled or disabled for an entire crate with the
 > [`-C target-feature`] flag.
 
+## The `track_caller` attribute
+
+The `track_caller` attribute may be applied to any function with [`"Rust"` ABI][rust-abi]. When
+applied to functions and methods in trait declarations, the attribute applies to all
+implementations. If the trait provides a default implementation with the attribute, then the
+attribute also applies to override implementations.
+
+When applied to a function in an `extern` block the attribute must also be applied to any linked
+implementations, otherwise undefined behavior results. When applied to a function which is made
+available to an `extern` block, the declaration in the `extern` block must also have the attribute,
+otherwise undefined behavior results.
+
+### Behavior
+
+Applying the attribute to a function `f` allows code within `f` to observe the [`Location`] of the
+"topmost" tracked call that led to `f`'s invocation. At the point of observation, an implementation
+behaves as if it walks up the stack from `f`'s frame to find the nearest frame of an *unattributed*
+function `outer`, and it returns the [`Location`] of the tracked call in `outer`.
```

Comment:
I feel like this paragraph should mention the "hint-like" character of this attribute. It reads to me as though the behavior is guaranteed. I think just using the word "hint" would suffice, perhaps like "Note however that this result is a "hint"; in some scenarios we may halt the walk early", or even just saying "However, be sure to read the Limitations section for some important caveats."

Reply:
Good call, think I've addressed this.